tools/GdkTestRunner/Modules/UnityModule.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using GdkTestRunner.Model;
+using Newtonsoft.Json.Linq;
+using NLog;
+using UnityPaths;
+
+namespace GdkTestRunner.Modules
+{
+    [TestModuleIdentifier("unity", ModuleDocumentation)]
+    public class UnityModule : BaseModule
+    {
+        public override string Name { get; }
+
+        private const string ModuleDocumentation = @"
+This test module allows for you to test Unity projects.
+
+The canonical configuration JSON for this module looks like the following:
+
+        {
+            ""type"": ""unity"",
+            ""name:"": ""Unity GDK Editmode Tests"",
+            ""parameters"": {
+                ""project_path"": ""../workers/unity"",
+                ""test_platform"": ""editmode""
+            }
+        }
+
+where:
+
+    'type'         - the type of this module, unity
+    'name'         - the name of this test, used in logging. This should be unique.
+    'project_path' - the path, relative to the JSON file, to the unity project folder
+
+The test results file will be placed in <cwd>/logs/name-as-kebab-case-test_platform.xml.
+The Unity log file will be placed in <cwd>/logs/name-as-kebab-case-test_platform.log.
+";
+
+        private readonly string unityProjectPath;
+        private readonly string unityTestPlatform;
+        private readonly string logfilePath;
+        private readonly string testResultsPath;
+
+        private readonly Logger logger;
+
+        public UnityModule(GdkTestRunnerOptions options, JToken jsonContext) : base(options, jsonContext)
+        {
+            var parsedJson = jsonContext.ToObject<UnityModuleDefinition>();
+
+            Name = parsedJson.Name;
+            unityProjectPath = Path.Combine(Path.GetDirectoryName(options.ConfigurationFilePath),
+                parsedJson.UnityDefinition.UnityProjectPath);
+            unityTestPlatform = parsedJson.UnityDefinition.TestPlatform;
+
+            if (!ValidateTestPlatform())
+            {
+                throw new ArgumentException($"Unknown test platform: {unityTestPlatform}");
+            }
+
+            var formattedName = TitleCaseToKebabCase(parsedJson.Name);
+
+            logfilePath = GetLogFilePath($"{formattedName}-{unityTestPlatform}.log");
+            testResultsPath = GetLogFilePath($"{formattedName}-{unityTestPlatform}-results.xml");
+
+            logger = LogManager.GetCurrentClassLogger();
+        }
+
+        protected override void BeforeRun()
+        {
+            CleanLibraryAndTempFolders();
+        }
+
+        protected override bool InternalRun()
+        {
+            var currentDirectory = Environment.CurrentDirectory;
+            Environment.CurrentDirectory = unityProjectPath;
+
+            var unityPath = GetUnityExePath(Paths.TryGetUnityPath());
```

Comment:
would it be better to use the `RunUnity` project here to avoid duplicating code?

Reply:
Well its a binary so I'd have to invoke `dotnet run RunUnity` to invoke Unity which is another layer of abstraction

Reply:
but, you wouldn't duplicate code :P